edit/networking.k8s.io.networkpolicy/PolicyRule.vue

```diff
@@ -0,0 +1,103 @@
+<script>
+
+import PolicyRulePort from '@/edit/networking.k8s.io.networkpolicy/PolicyRulePort';
+import ArrayListGrouped from '@/components/form/ArrayListGrouped';
+import { _EDIT } from '@/config/query-params';
+import PolicyRuleTarget from '@/edit/networking.k8s.io.networkpolicy/PolicyRuleTarget';
+
+export default {
+  components: {
+    ArrayListGrouped, PolicyRulePort, PolicyRuleTarget
+  },
+  props:      {
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      },
+    },
+    mode: {
+      type:    String,
+      default: _EDIT,
+    },
+    type: {
+      type:    String,
+      default: 'ingress'
+    },
+    namespace: {
+      type:    String,
+      default: ''
+    },
+    allPods: {
+      type:    Array,
+      default: () => {
+        return [];
+      },
+    },
+    allNamespaces: {
+      type:    Array,
+      default: () => {
+        return [];
+      },
+    },
+  },
+  methods: {
+    onAddPort() {
+      if (this.$refs.lastPort?.focus) {
+        this.$refs.lastPort.focus();
```

Comment:
I didn't see this or the one below working. These are convenience methods though, so worst case they can be removed

Reply:
I removed these